client/src/app/services/headerInterceptor.ts

```diff
@@ -0,0 +1,27 @@
+import {
+    HttpEvent,
+    HttpInterceptor,
+    HttpHandler,
+    HttpRequest,
+} from '@angular/common/http';
+import { User } from '../models/user';
+import { AuthenticationService } from './authentication.service'
+
+import { Observable } from 'rxjs';
+import { Injectable } from '@angular/core';
+@Injectable() 
+export class AddHeaderInterceptor implements HttpInterceptor {
+    public currentUser: User;
+    constructor(private as: AuthenticationService) { }
+    intercept(req: HttpRequest<any>, next: HttpHandler): Observable<HttpEvent<any>> {
+
+        let token = '';
+        if(this.as.currentUserValue !== null){
+            token = this.as.currentUserValue.token;
+        };
+        // Clone the request to add the new header
```

Comment:
you need to set those headers only if 
`currentUser && currentUser.token`

also we can check for proper ApiUrl
`request.url.startsWith(config.YourApiUrl); // teruen boolean`

Reply:
Done but i used 
`currentUserValue && currentUserValue.token` 
Should i created config file here and put there 
`apiUrl = '/api` ?

Reply:
`import { environment as config } from '../../environments/environment';`
and put 
`  apiUrl: 'http://127.0.0.1:PUT_YOUR_SERVERS_PORT_HERE/api',`
into environment.ts and environment.prod.ts